src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -876,6 +923,7 @@ void PendingDataset::StartLeader(void)
 {
     UpdateDelayTimer(mLocal, mLocalTime);
     mNetwork = mLocal;
+    mNetwork.Store();
```

Comment:
`mNetwork` and `mLocal` are the same here, so I think this ok.  But in general, I think we should be storing the `mLocal` copy.